docs/concurrency.md

```diff
@@ -125,12 +125,13 @@ It is known that, when run together, _at most one_ of A and B can go into an inf
 We need to implement a function `firstResult()` that will run A and B concurrently and wait for the first result that is returned by either of them.
 The function needs to return this first result.
 
-It is clear that this task cannot be simulated on a single-CPU machine that cannot actually run two processes concurrently and wait for the first result.
-Regardless of how we implement `firstResult()`, a single-CPU machine will have to run at least one of the processes A and B.
-Sometimes, a process will go into an infinite loop; in that case, the single-CPU machine will never finish computing `firstResult()`.
+Now, we claim that this task cannot be simulated on a single thread, because no program running on a single thread can actually run two processes concurrently and wait for the first result.
+Here is why:
+Regardless of how we implement `firstResult()`, a single-thread program will have to run at least one of the processes A and B on that single thread.
+Sometimes, that process will go into an infinite loop; in that case, the single thread will be infinitely blocked, and our program will never finish computing `firstResult()`.
 
-So, when implemented on a single-CPU machine, `firstResult()` must be a _partial function_ (it will sometimes fail to return a value).
-However, on a multi-CPU machine, we can implement `firstResult()` as a _total function_ (it will always return a result),
+So, when implemented on a single thread, `firstResult()` is a _partial function_ (it will sometimes fail to return a value).
+However, on a multi-thread machine, we can implement `firstResult()` as a _total function_ (it will always return a result),
```

Comment:
machine? You moved away from hardware to operating systems library (threads library) in your discussion.